internal/query/result_range_test.go

```diff
@@ -0,0 +1,363 @@
+//go:build go1.22 && goexperiment.rangefunc
+
+package query
+
+import (
+	"context"
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ydb-platform/ydb-go-genproto/protos/Ydb"
+	"github.com/ydb-platform/ydb-go-genproto/protos/Ydb_Query"
+	"go.uber.org/mock/gomock"
+
+	"github.com/ydb-platform/ydb-go-sdk/v3/internal/xtest"
+)
+
+func TestResultRange(t *testing.T) {
```

Comment:
need additional tests for behavior if user break loop.
I expect about result will be closed.